### DISCUSSION TITLE
I would like to use multiple schema files

### DISCUSSION THREAD

**Discussion by mugishap** [prisma/prisma#20878] [Q&A] [answered]
## Problem

I have a very large project with many models(100+). It is a very big disadvantage finding that I can't be able to view models very well because they are very many.

## Suggested solution

I think there should  be a way I can setup multiple schema.prisma files. Or even have a .prisma file that can import other models from other .prisma files and then specify the path to your schemas.

**Comment by gmwill934** [comment]
Hi, you could check out [prisma-import ](https://github.com/ajmnz/prisma-import)

**Comment by hussamkhatib** [comment]
Check out [Schemix](https://github.com/ridafkih/schemix) as well

**Comment by ludralph** [comment]
Hi @mugishap 👋 

Thank you for raising this question. We do have a feature request to support splitting Prisma schema into multiple files. It would be helpful if you could add your use case to the issue and leave a 👍 on the issue to help our engineers prioritize it.

- https://github.com/prisma/prisma/issues/2377

The workaround suggested by both @gmwill934 and @hussamkhatib should work for you.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise, please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ludralph** [reply]
Hey @mugishap 

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Comment by sorenbs** [comment, marked as answer]
This has now been implemented: https://www.prisma.io/blog/organize-your-prisma-schema-with-multi-file-support

**Reply by oadrianrabelo** [reply]
it works for different datasources?

**Reply by Twilight-Techy** [reply]
All my models were previously declared in one prisma file, but on separating them and running npx prisma migrate, I'm getting warnings saying I'm about to drop several of my tables with existing rows of data. I didn't alter any of my models, just separated them into different files. Is there a way I can migrate my my database without losing data?
